etna/pipeline/pipeline.py

```diff
@@ -0,0 +1,59 @@
+from copy import deepcopy
+from typing import Iterable
+
+from etna.datasets import TSDataset
+from etna.models.base import Model
+from etna.transforms.base import Transform
+
+
+class Pipeline:
+    """Pipeline of transforms with a final estimator."""
+
+    def __init__(self, model: Model, transforms: Iterable[Transform] = [], horizon: int = 1):
+        """
+        Create instance of Pipeline with given parameters.
+
+        Parameters
+        ----------
+        model:
+            Instance of the etna Model
+        transforms:
+            Sequence of the transforms
+        horizon:
+            Number of timestamps in the future for forecasting
+        """
+        self.model = model
+        self.transforms = transforms
+        self.horizon = horizon
+        self.ts = None
+
+    def fit(self, ts: TSDataset) -> "Pipeline":
+        """ Fit the Pipeline.
+        Fit and apply given transforms to the data, then fit the model on the transformed data.
+
+        Parameters
+        ----------
+        ts:
+            Dataset with timeseries data
+        Returns
+        -------
+        self :
+            Fitted Pipeline instance
+        """
+        ts = deepcopy(ts)
```

Comment:
May be we can use `ts.copy(deep=True)` in this case.

Reply:
TSDataset does not have method copy

Reply:
In this case we do not need deep copy because we work with TSDataset (has `raw_df`) and not with pd.DataFrame